addOns/quickstart/src/main/java/org/zaproxy/zap/extension/quickstart/AttackThread.java

```diff
@@ -44,207 +44,243 @@
 import org.zaproxy.zap.model.Target;
 
 public class AttackThread extends Thread {
-	
-	public enum Progress {notstarted, started, spider, ascan, failed, complete, stopped }
-	
-	private ExtensionQuickStart extension;
-	private URL url;
+
+    public enum Progress {
+        notstarted, started, spider, ajaxspider, ascan, failed, complete, stopped
+    }
+
+    private ExtensionQuickStart extension;
+    private URL url;
     private HttpSender httpSender = null;
+    private PlugableSpider plugableExplorer;
     private boolean stopAttack = false;
+    private boolean useStdSpider;
 
     private static final Logger logger = Logger.getLogger(AttackThread.class);
 
-	public AttackThread(ExtensionQuickStart ext) {
-		super("ZAP-QuickStart-AttackThread");
-		this.extension = ext;
-	}
-	
-	public void setURL(URL url) {
-		this.url = url;
-	}
-	
-	@Override
-	public void run() {
-		stopAttack = false;
+    public AttackThread(ExtensionQuickStart ext, boolean useStdSpider) {
+        super("ZAP-QuickStart-AttackThread");
+        this.extension = ext;
+        this.useStdSpider = useStdSpider;
+    }
+
+    public void setURL(URL url) {
+        this.url = url;
+    }
+
+    public void setPlugableExplorer(PlugableSpider plugableExplorer) {
+        this.plugableExplorer = plugableExplorer;
+    }
+
+    @Override
+    public void run() {
+        stopAttack = false;
         try {
```

Comment:
We should add a finally block to ensure a consistent state of the attack (in case of errors calling the AJAX Spider).